src/rgw/rgw_rados.cc

```diff
@@ -426,7 +426,7 @@ void RGWZoneGroup::post_process_params()
   for (map<string, RGWZone>::iterator iter = zones.begin(); iter != zones.end(); ++iter) {
     RGWZone& zone = iter->second;
     zone.log_data = log_data;
-    zone.log_meta = (is_master && zone.id == master_zone);
+    zone.log_meta = (is_master && zone.id == master_zone && zone.name != default_zone_name);
```

Comment:
i worry that this would cause problems for default configurations that are later extended with more zones (that's possible to do since #10477). ideally this check would be based on `num_zonegroups == 1 && num_zones == 1`, but that's tricky to do in a `RGWZoneGroup` member function during a `zone[group] modify` command

maybe it would be better to take the approach from #15613, and add logic to `RGWRados::need_to_log_metadata()` that could inspect the `current_period` to count zonegroups/zones?